[CI] CircuitBreakingException in qa:google-cloud-storage "10_repository/Register a repository with a non existing client" test

Failure: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+g1gc/6050/console

Error:
```
java.lang.AssertionError: Failure at [repository_gcs/10_repository:191]: the error message was expected to match the provided regex but didn't
Expected: repository_exception
     but: was "{root_cause=[{type=circuit_breaking_exception, reason=[parent] Data too large, data for [<http_request>] would be [510357400/486.7mb], which is larger than the limit of [510027366/486.3mb], bytes_wanted=510357400, bytes_limit=510027366, stack_trace=CircuitBreakingException[[parent] Data too large, data for [<http_request>] would be [510357400/486.7mb], which is larger than the limit of [510027366/486.3mb]]\n\tat org.elasticsearch.indices.breaker.HierarchyCircuitBreakerService.checkParentLimit(HierarchyCircuitBreakerService.java:257)\n\tat org.elasticsearch.common.breaker.ChildMemoryCircuitBreaker.addEstimateBytesAndMaybeBreak(ChildMemoryCircuitBreaker.java:128)\n\tat org.elasticsearch.rest.RestController.dispatchRequest(RestController.java:231)\n\tat org.elasticsearch.rest.RestController.tryAllHandlers(RestController.java:336)\n\tat org.elasticsearch.rest.RestController.dispatchRequest(RestController.java:173)\n\tat 
[...]
```

I couldn't reproduce locally so far:
```
REPRODUCE WITH: ./gradlew :plugins:repository-gcs:qa:google-cloud-storage:integTestRunner \
  -Dtests.seed=76E4624F4840A694 \
  -Dtests.class=org.elasticsearch.repositories.gcs.GoogleCloudStorageRepositoryClientYamlTestSuiteIT \
  -Dtests.method="test {yaml=repository_gcs/10_repository/Register a repository with a non existing client}" \
  -Dtests.security.manager=true \
  -Dtests.jvm.argline="-XX:-UseConcMarkSweepGC -XX:+UseG1GC" \
  -Dtests.locale=he \
  -Dtests.timezone=Etc/GMT-5
```

--- a/buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy
+++ b/buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy
@@ -331,6 +331,12 @@ class ClusterFormationTasks {
         }
         // increase script compilation limit since tests can rapid-fire script compilations
         esConfig['script.max_compilations_rate'] = '2048/1m'
+        // Temporarily disable the real memory usage circuit breaker. It depends on real memory usage which we have no full control
+        // over and the REST client will not retry on circuit breaking exceptions yet (see #31986 for details). Once the REST client
+        // can retry on circuit breaking exceptions, we can revert again to the default configuration.
+        if (node.nodeVersion.major >= 7) {
+            esConfig['indices.breaker.total.use_real_memory'] = false
+        }
         esConfig.putAll(node.config.settings)
 
         Task writeConfig = project.tasks.create(name: name, type: DefaultTask, dependsOn: setup)

